### PR TITLE
Coralogix dashboard

### PR DESCRIPTION
**Pull request by OrNovo**
Adding resource and data-source coralogix-dashboard

### CODE REVIEW

File: coralogix/resource_coralogix_dashboard.go [hunk 1]
```diff
@@ -0,0 +1,1663 @@
...
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "Dashboard name.",
+		},
+		"description": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Dashboard description.",
+		},
+		"layout": {
```
Comment (oded-dd) [inline coralogix/resource_coralogix_dashboard.go:1236]: We need to be careful here, if the schema is breaking due to a change we did, we need to make sure we'll have a way to support backward compatible.

Comment (oded-dd) [reply coralogix/resource_coralogix_dashboard.go:1236]: Also, don't we want to support file structure like we did in the enrichment (e.g. file = <layout.json>)

File: coralogix/resource_coralogix_dashboard_test.go [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+				return fmt.Errorf("dashboard still exists: %s", rs.Primary.ID)
+			}
+		}
+	}
+
+	return nil
+}
+
+func testAccCoralogixResourceDashboard() string {
+	return fmt.Sprintf(`resource "coralogix_dashboard" test {
```
Comment (iravid) [inline coralogix/resource_coralogix_dashboard_test.go:138]: Wow this is really cool

File: examples/dashboard/main.tf [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+              columns {
+                field           = "coralogix.metadata.subsystemName"
+                order_direction = "Unspecified"
+              }
+              columns {
+                field           = "coralogix.metadata.sdkId"
+                order_direction = "Unspecified"
+              }
+              columns {
+                field           = "textObject.textObject.textObject.log_obj.e2e_test.config"
```
Comment (iravid) [inline examples/dashboard/main.tf:138]: @OrNovo i think the repeated textObject prefix can be dropped - could you check with the team?

File: coralogix/resource_coralogix_dashboard_test.go (hunk 2 above)
Comment (OrNovo) [reply coralogix/resource_coralogix_dashboard_test.go:138]: Thanks 🙏

**Review by oded-dd — DISMISSED** [review]

**Review by oded-dd — APPROVED** [review]

**Pull request merged**